src/configs/time.rs

```diff
@@ -0,0 +1,28 @@
+use crate::config::{ModuleConfig, RootModuleConfig, SegmentConfig};
+
+use ansi_term::{Color, Style};
+use starship_module_config_derive::ModuleConfig;
+
+#[derive(Clone, ModuleConfig)]
+pub struct TimeConfig<'a> {
+    pub time: SegmentConfig<'a>,
+    pub use_12hr: bool,
+    pub format: Option<&'a str>,
+    pub style: Style,
+    pub disabled: bool,
+}
+
+impl<'a> RootModuleConfig<'a> for TimeConfig<'a> {
+    fn new() -> Self {
+        TimeConfig {
+            time: SegmentConfig {
+                value: "",
+                style: None,
+            },
```

Comment:
Sorry for the inconvenience but styling an undocumented segment is currently under discussion in #472. This syntax shouldn't be used until we get that issue resolved.

You can refer to b0bf17277e868026efe678d81915cc4941aa9e50 to drop it.